### PR TITLE
Improved payload3 support for algorand

### PR DESCRIPTION
**Pull request by jumpsiegel**
After some feedback from the user community, I adjusted the payload3 API on algorand to more sense

changes:

    Add payload3 support into typescript
    Add tests for payload3 in typescript
    Change the contract API for payload3 so that the destination is the appId instead of the app address  and no longer part of the data
    add a testApp into devnet

---
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1158)

### CODE REVIEW

File: sdk/js/src/algorand/Algorand.ts [hunk 1]
```diff
@@ -815,7 +813,15 @@ export async function _submitVAAAlgorand(
 
     // The receiver needs to be optin in to receive the coins... Yeah, the relayer pays for this
 
-    const addr = encodeAddress(hexToUint8Array(parsedVAA.get("ToAddress")));
+    let aid = 0;
+    let addr;
+
+    if ((parsedVAA.get("ToChain") == 8) && (parsedVAA.get("Type") == 3)) {
```
Comment (panoel) [inline sdk/js/src/algorand/Algorand.ts:819]: if ((parsedVAA.get("ToChain") === ChainIDAlgorand) && (parsedVAA.get("Type") === 3)) {

Comment (panoel) [inline sdk/js/src/algorand/Algorand.ts:819]: if ((parsedVAA.get("ToChain") === ChainIDAlgorand) && (parsedVAA.get("Type") === 3)) {

File: sdk/js/src/token_bridge/__tests__/integration.ts [hunk 2]
```diff
@@ -3085,5 +3085,98 @@ describe("Integration Tests", () => {
...
+          const aa: string = uint8ArrayToHex(decTbAddr);
+
+          const client: algosdk.Algodv2 = getAlgoClient();
+          const tempAccts: Account[] = await getTempAccounts();
+          const numAccts: number = tempAccts.length;
+          expect(numAccts).toBeGreaterThan(0);
+          const algoWallet: Account = tempAccts[0];
+
+          const Fee: number = 0;
+          var testapp: number = 8;
```
Comment (panoel) [inline sdk/js/src/token_bridge/__tests__/integration.ts:3104]: Just out of curiosity, what is this?

Comment (panoel) [inline sdk/js/src/token_bridge/__tests__/integration.ts:3104]: Just out of curiosity, what is this?

File: sdk/js/src/token_bridge/__tests__/integration.ts [hunk 3]
```diff
@@ -3085,5 +3085,98 @@ describe("Integration Tests", () => {
...
+              signedVaa.vaaBytes
+            )
+          ).toBe(true);
+          const wafter = await getBalance(client, getApplicationAddress(testapp), BigInt(0));
+          console.log(
+            "test app wallet after:",
+            wafter
+          );
+
+          expect(BigInt(wafter - wbefore) == BigInt(100));
```
Comment (panoel) [inline sdk/js/src/token_bridge/__tests__/integration.ts:3171]: Use ===

Comment (panoel) [inline sdk/js/src/token_bridge/__tests__/integration.ts:3171]: Use ===

**Review by panoel — APPROVED** [review]

**Review by panoel — APPROVED** [review]

**Review by kcsongor — APPROVED** [review]

**Pull request merged**